pkg/client/sockets.go

```diff
@@ -66,6 +66,10 @@ func SocketURL(socket string) string {
 }
 
 // DialSocket dials the given unix socket and returns the resulting connection
-func DialSocket(c context.Context, socketName string) (*grpc.ClientConn, error) {
-	return grpc.DialContext(c, SocketURL(socketName), grpc.WithInsecure(), grpc.WithNoProxy())
+func DialSocket(ctx context.Context, socketName string) (*grpc.ClientConn, error) {
+	return grpc.DialContext(ctx, SocketURL(socketName),
+		grpc.WithInsecure(),
+		grpc.WithNoProxy(),
+		grpc.WithBlock(),
+	)
```

Comment:
The only thing that actually changed in this block is the addition of `grpc.WithBlock()`.